sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala

```diff
@@ -200,8 +200,7 @@ class RocksDBFileManager(
       case 2 =>
         new StateStoreChangelogWriterV2(fm, changelogFile, codec)
       case _ =>
-        throw new IllegalArgumentException(s"Failed to find changelog writer for " +
-          s"version=$changelogVersion")
+        throw QueryExecutionErrors.invalidChangeLogWriterVersion(changelogVersion)
```

Comment:
I guess no easy way to test either of these error classes right ?

Reply:
We can manually write a "V666" changelog file to the checkpoint directory. Though I am also ok if this is not tested.

Reply:
Yea - same for OOM right. probably no easy to test this

Reply:
I would prefer for the version test to be added. But will let @riyaverm-db / @chaoqin-li1123 decide which route they prefer to take.

Reply:
Not that I can think of. I think this should be fine since this is just adding a layer of classification before throwing uncategorized error.

Reply:
We can see when we call getChangeLogVersion it only returns 1 or 2 based on a boolean useColumnFamilies. Versioning is not added to change log file format so we cannot add this test @anishshri-db.

Reply:
Ok yea sg - this is an artifact of the version being added in the non-v0 version. but i think thats fine and the change we have should be fine